markdown/developers/tutorials/dsl2_modules_tutorial.md

```diff
@@ -11,414 +11,298 @@ If you create a new module with the goal of contributing the code to *nf-core*,
+    1. [`./modules/fastqc/main.nf`](https://github.com/nf-core/modules/blob/master/modules/fastqc/main.nf)
 
-```bash
-nf-core modules lint --tool fgbio/fastqtobam .
-```
+        This is the main script containing the `process` definition for the module. You will see an extensive number of `TODO` statements to help guide you to fill in the appropriate sections and to ensure that you adhere to the guidelines we have set for module submissions.
 
-You will expect no test failed, as shown in figure below:
+    2. [`./modules/fastqc/meta.yml`](https://github.com/nf-core/modules/blob/master/modules/fastqc/meta.yml)
 
-![lint](/assets/markdown_assets/developers/dsl2_modules_tutorial/dsl2-mod_04_lint_module.png)
+        This file will be used to store general information about the module and author details - the majority of which will already be auto-filled. However, you will need to add a brief description of the files defined in the `input` and `output` section of the main script since these will be unique to each module.
 
-## Test your code
+    3. [`./tests/modules/fastqc/main.nf`](https://github.com/nf-core/modules/blob/master/tests/modules/fastqc/main.nf)
 
-Once your code is polished, following any suggestions from linting, you should test the code and make sure everything works as expected.
-This can also be done automatically, using the [pytest-worfklow](https://pytest-workflow.readthedocs.io/en/stable/) tool.
+        Every module MUST have a test workflow. This file will define one or more Nextflow `workflow` definitions that will be used to unit test the output files created by the module. By default, one `workflow` definition will be added but please feel free to add as many as possible so we can ensure that the module works on different data types / parameters e.g. separate `workflow` for single-end and paired-end data.
 
-### Create a test workflow
+        Minimal test data required for your module may already exist within the [nf-core/modules repository](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config), in which case you may just have to change a couple of paths in this file - see the [Test data](#test-data) section for more info and guidelines for adding new standardised data if required.
 
-As described above, *nf-core tools* has created already the following files
+    4. [`./tests/modules/fastqc/nextflow.config`](https://github.com/nf-core/modules/blob/master/tests/modules/amps/nextflow.config)
 
-```bash
-tests/modules/fgbio
-└── fastqtobam
-    ├── main.nf
-    └── test.yml
-```
+        Some modules MAY require additional parameters added to the test command to successfully run. These can be specified with an `ext.args` variable within the process scope of the `nextflow.config` file that exists alongside the test files themselves (and is automatically loaded when the test workflow `main.nf` is executed).
```

Comment:
this part is really crucial, and it's probably one of the most nested feature of the new template.
I understand it's well explained in the module guidelines page, but I also think one should go through it here again with an example that refers to the specific use-case that's used for the tutorial here.
We can't take this for granted, and a user without much experience will find this part difficult to navigate through without a concrete example within the context of this page.